src/basic/path-util.c

```diff
@@ -633,7 +646,9 @@ int find_binary(const char *name, char **ret) {
                 if (!path_is_absolute(element))
                         continue;
 
-                j = path_join(element, name);
+                path = prefix_roota(root, element);
+
+                j = path_join(path, name);
```

Comment:
so, the access() further down will resolve symlinks relative to the current mount namespace. Which is wrong. You need to use chase_symlinks() i guess, to find the binary, so that symlinks are resolved relative to the specified root.